### PR TITLE
Add tool to try Mapael

### PR DESCRIPTION
**Pull request by neveldo**
Hello @Indigo744 ,

Here is an improved version for the test page : https://www.vincentbroute.fr/mapael/try.html . Added : 

- A way to see the hovered area ID, pageX, pageY, map X and map Y.
- A button to export the map to JSFiddle
- A button to save the map into a file
- A way to upload a CSV file that contains areas/plots data to be displayed on the map

It can still be improved (add a way to chose Raphael / Mapael version, retrieve the maps from Github, etc), but I think that it's not bad for a first version. I hope you will like it :+1:)

Related issue : https://github.com/neveldo/mapael-documentation/issues/6#issuecomment-340720870

### CODE REVIEW

**Comment by Indigo744** [comment]
This is so cool! I love it :smile:

Some small remarks regarding the bottom information: 

1. Move `Hovered area ID` info at the beginning of string, and add a line break after it so the mouse coordinates are on a second line
2. When clicked on the map to fix the value, maybe add a reticle (I mean only drawing two lines that are crossing on the point). This could be done maybe with jQuery by simply adding two absolutely positioned element?
3. Add lat/long coordinates if they are available (check first if a call return something else than (0, 0) which would mean the function isn't implemented)
4. Duplicate these information also at top of the map, as for some big map it's hard to scroll down and up to check values... Or maybe add a `max-height` value to the map container so it's always visible (maybe use the `vh` unit, like `70vh` to be sure it's not larger than the viewport.

**Comment by Indigo744** [comment]
Also, the `export to JSFiddle` button generates a weird URL with mootools as dependency... e.g. http://jsfiddle.net/api/post/mootools/1.2/dependencies/more/

**Comment by neveldo** [comment]
Glad you like it ! Here we go for the reticle : https://www.vincentbroute.fr/mapael/try.html . I have also fixed the form action for JSFiddle export.

However, how could we retrieve a latitude and longitude from (x,y) coordinates ? getCoords() is just a one-way function :(

**Comment by Indigo744** [comment]
Aww right... I didn't though about that... 😞

**Comment by Indigo744** [comment]
About the reticle: why should I select the `Spot a point on click` option? ~~It would be better to keep it simple: if I click (and not dragging), add the reticle.~~ It is actually good to avoid issue with onClick map event. However, dragging should be detected, right?

**Comment by Indigo744** [comment]
Also, I still think we should limit the height of the map to always have it inside the viewport. No?

**Comment by neveldo** [comment]
> It would be better to keep it simple: if I click (and not dragging), add the reticle.

I added this checkbox having in mind the users who need to test regular click events on a map (for instance, area coloration on click, zoom on click, etc). It would be weird for them to have a reticle drawn on the map whenever they click on it !

> However, dragging should be detected, right?

Yes, of course !

> Also, I still think we should limit the height of the map to always have it inside the viewport. No?

Done ! I fixed the height and add an `overflow-y:auto;` to the container. It looks better, however, it seems that the scrollbar causes two bugs : 

- The map overflow the right boundary of the container
- The reticle is not well positionned

(you can see them with the maps of France for instance. Maybe I should be another solution without using a scrollbar ?

**Comment by Indigo744** [comment]
In Firefox, I don't even see the scrollbars actually:
image

File: try.html [hunk 1]
```diff
@@ -156,35 +163,56 @@
...
+
+        </div>
 
-                <label for="configuration">Map configuration</label>
+        <!-- Map extra info pannel -->
+        <div class="col-sm-6">
+            <div class="mapExtraInfo">
+                Hovered area ID : <span class="area">-</span>
+                <br />pageX : <span class="pageX">0</span>, pageY : <span class="pageY">0</span> - X : <span class="x">0</span>, Y : <span class="y">0</span>
+                <br /><label for="spotPoint">Spot a point on click</label> <input type="checkbox" class="spotPoint" name="spotPoint" />
```
Comment (Indigo744) [inline try.html]: Add `id="spotPoint"` to checkbox input in order to register the click on label.

**Comment by neveldo** [comment]
Strange, as I have only used regular CSS : 

```
            height:70vh;
            overflow-y:auto;
```

I will check it asap !

File: try.html [hunk 2]
```diff
@@ -0,0 +1,766 @@
...
+
+        <!-- Map url / file / CSV pannel -->
+        <div class="col-sm-6">
+
+            <div class="form-group">
+
+                <div class="row">
+                    <div class="col-sm-4">
+                        <label for="mapSelector">Map</label>
+                        <select class="form-control" id="mapSelector" name="mapSelector"></select>
```
Comment (Indigo744) [inline try.html:178]: Clicking on the Map label seems to trigger the focus on the select, but it would be best if it opens it as well?

**Comment by Indigo744** [comment]
In another project, I had an image that needed to be responsive, but I wanted to apply a max-width *and* a max-height. I had it working at last using this [SO answer](https://stackoverflow.com/a/33420186/8861729). Don't know it it could apply here.

**Comment by Indigo744** [comment]
In the JSFiddle export, could the JS code for Mapael appear in the Javascript part? So people can easily modifiy it :smile:

**Comment by neveldo** [comment]
It seems that I have not uploaded the last version, you should now see the scrollbar (and the issue on maps of France. It seems there is no problem on argentina map for instance, I need to investigate).

> Add id="spotPoint" to checkbox input in order to register the click on label.

Done

> Clicking on the Map label seems to trigger the focus on the select, but it would be best if it opens it as well?

Done

> In the JSFiddle export, could the JS code for Mapael appear in the Javascript part? So people can easily modifiy it :smile:

Unfortunately, no :( . There is a known bug with the POST feature on JSFiddle : the external libraries are loaded into random order and raise errors if the order is wrong. See https://stackoverflow.com/questions/41927911/post-to-jsfiddle-shows-external-resources-in-wrong-order

**Comment by neveldo** [comment]
I have replaced the `overflow-y:auto;` by `overflow-y:scroll;` on the `.mapcontainer` div, and in seems to solve the width issue.

**Comment by Indigo744** [comment]
The issue highlighted by SO regards external resources. But can we simply keep the libraries in the HTML script tags, like now, but only put the init code in JS part (with one important thing: to select the `onDomready` load type in JS options)?

Like this:
image

**Comment by neveldo** [comment]
Hello @Indigo744 ,

Thanks, I didn't know about this option from JSFiddle, it should be ok now :)

**Comment by Indigo744** [comment]
Another option would have been to use the `$.ready()` of jQuery, but this one is easier :smile:

**Comment by neveldo** [comment]
You are right, there is a little less code displayed in JSFiddle this way !

**Comment by neveldo** [comment]
Hello @Indigo744 ,

Do you think that the tool is ok to be merged now, or do you have in mind some more improvements ?

**Comment by Indigo744** [comment]
Yes! It's a great tool 😃

**Pull request merged**